# Table of Contents

**1. Import Libraries**  

**2. Import Dataset** 

**3. Data Cleaning Steps**  
>>3.a Check for Mixed Datatypes, Rename Columns, Change Column Data types if needed        
>>3.b Check for Missing Values   
>>3.c Check for Duplicates  

**4. Statistical Analysis**   


**5. Export Cleaned Dataset**    



## 1. Import Libraries

In [96]:
# Import libraries
import pandas as pd
import numpy as np
import os

## 2. Import Dataset

In [97]:
# Create a Path variable
path=r'C:\Users\sands\Documents\03-2023 Airbnb New York'

# Read the Airbnb Listings CSV file
df_list = pd.read_csv(os.path.join(path,'02 Data', 'Original Data','NewYorkABListings.csv'),index_col=False,
                      dtype={'license': 'string'}).fillna('')

In [98]:
# Check the imported dataset
df_list.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41533 entries, 0 to 41532
Data columns (total 18 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              41533 non-null  int64  
 1   name                            41533 non-null  object 
 2   host_id                         41533 non-null  int64  
 3   host_name                       41533 non-null  object 
 4   neighbourhood_group             41533 non-null  object 
 5   neighbourhood                   41533 non-null  object 
 6   latitude                        41533 non-null  float64
 7   longitude                       41533 non-null  float64
 8   room_type                       41533 non-null  object 
 9   price                           41533 non-null  int64  
 10  minimum_nights                  41533 non-null  int64  
 11  number_of_reviews               41533 non-null  int64  
 12  last_review                     

## 3. Data Cleaning Steps

**3.a Check for Mixed Datatypes**

In [99]:
# Check for Mixed Datatypes
for col in df_list.columns.tolist():
  mixed = (df_list[[col]].applymap(type) != df_list[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_list[mixed]) > 0:
    print (col)

reviews_per_month


In [100]:
# Rename the columns, Re-Assign correct datatypes to the columns and Drop the empty license column
df_list =df_list.astype({'id':'string',
                         'name':'string',
                         'host_id':'string',
                          'host_name':'string',
                          'neighbourhood_group':'string',
                          'neighbourhood':'string',
                          'room_type':'string',
                          'last_review':'datetime64[ns]'}).rename(columns={'id': 'Id',
                 'name': 'Listing Name',
                 'host_id': 'Host Id',
                 'host_name': 'Host Name',
                 'neighbourhood_group': 'Neighbourhood Group', 
                 'neighbourhood': 'Neighbourhood',
                 'latitude': 'Latitude',
                 'longitude': 'Longitude', 
                 'room_type': 'Listing Type', 
                 'price': 'Price', 
                 'minimum_nights': 'Minimum Nights', 
                 'number_of_reviews': 'Total Reviews', 
                 'last_review': 'Last Review Date',
                 'reviews_per_month': 'Reviews per Month',
                 'calculated_host_listings_count': 'Host Listings Count', 
                 'availability_365': 'Future Availability', 
                 'number_of_reviews_ltm': 'Reviews in Last 12 Months'}).drop(columns=['license'])  

In [101]:
df_list.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41533 entries, 0 to 41532
Data columns (total 17 columns):
 #   Column                     Non-Null Count  Dtype         
---  ------                     --------------  -----         
 0   Id                         41533 non-null  int64         
 1   Listing Name               41533 non-null  string        
 2   Host Id                    41533 non-null  int64         
 3   Host Name                  41533 non-null  string        
 4   Neighbourhood Group        41533 non-null  string        
 5   Neighbourhood              41533 non-null  string        
 6   Latitude                   41533 non-null  float64       
 7   Longitude                  41533 non-null  float64       
 8   Listing Type               41533 non-null  string        
 9   Price                      41533 non-null  int64         
 10  Minimum Nights             41533 non-null  int64         
 11  Total Reviews              41533 non-null  int64         
 12  Last

**3.b Check for Missing Values**

In [102]:
df_list.isnull().sum()

Id                              0
Listing Name                    0
Host Id                         0
Host Name                       0
Neighbourhood Group             0
Neighbourhood                   0
Latitude                        0
Longitude                       0
Listing Type                    0
Price                           0
Minimum Nights                  0
Total Reviews                   0
Last Review Date             9393
Reviews per Month               0
Host Listings Count             0
Future Availability             0
Reviews in Last 12 Months       0
dtype: int64

**The Last Review Date column has 9393 missing values. Upon checking the Total Reviews column it was found that these are new listing that haven't been reviewed by any guest yet.**

**3.c Check for Duplicates**

In [103]:
# Create a subset of the duplicate values
df_dups = df_list[df_list.duplicated()]

In [104]:
#View the subset
df_dups

Empty DataFrame
Columns: [Id, Listing Name, Host Id, Host Name, Neighbourhood Group, Neighbourhood, Latitude, Longitude, Listing Type, Price, Minimum Nights, Total Reviews, Last Review Date, Reviews per Month, Host Listings Count, Future Availability, Reviews in Last 12 Months]
Index: []

**No duplicates found**

## 4. Statistical Analysis

In [105]:
df_list.describe()

Latitude     Longitude         Price  Minimum Nights  \
count  41533.000000  41533.000000  41533.000000    41533.000000   
mean      40.728292    -73.944526    221.978282       18.592204   
std        0.057145      0.055965    919.502236       30.699921   
min       40.500314    -74.249840      0.000000        1.000000   
25%       40.687750    -73.982410     80.000000        2.000000   
50%       40.723830    -73.953156    131.000000       10.000000   
75%       40.762200    -73.924990    220.000000       30.000000   
max       40.911380    -73.710870  98159.000000     1250.000000   

       Total Reviews  Host Listings Count  Future Availability  \
count   41533.000000         41533.000000         41533.000000   
mean       26.204994            20.626803           143.287771   
std        56.178847            68.874393           144.279510   
min         0.000000             1.000000             0.000000   
25%         1.000000             1.000000             0.000000   
50%         5.000000             1.000000            87.000000   
75%        25.000000             4.000000           312.000000   
max      1666.000000           487.000000           365.000000   

       Reviews in Last 12 Months  
count               41533.000000  
mean                    7.983700  
std                    18.563894  
min                     0.000000  
25%                     0.000000  
50%                     1.000000  
75%                     8.000000  
max                   992.000000

In [106]:
df_list.head()

Id                              Listing Name Host Id  Host Name  \
0  5136  Spacious Brooklyn Duplex, Patio + Garden    7378    Rebecca   
1  5203        Cozy Clean Guest Room - Family Apt    7490  MaryEllen   
2  5121                           BlissArtsSpace!    7356      Garon   
3  5178          Large Furnished Room Near B'way　    8967   Shunichi   
4  2595                     Skylit Midtown Castle    2845   Jennifer   

  Neighbourhood Group       Neighbourhood  Latitude  Longitude  \
0            Brooklyn         Sunset Park  40.66265  -73.99454   
1           Manhattan     Upper West Side  40.80380  -73.96751   
2            Brooklyn  Bedford-Stuyvesant  40.68535  -73.95512   
3           Manhattan             Midtown  40.76457  -73.98317   
4           Manhattan             Midtown  40.75356  -73.98559   

      Listing Type  Price  Minimum Nights  Total Reviews Last Review Date  \
0  Entire home/apt    275              21              3       2022-08-10   
1     Private room     75               2            118       2017-07-21   
2     Private room     60              30             50       2019-12-02   
3     Private room     68               2            559       2022-11-20   
4  Entire home/apt    175              30             49       2022-06-21   

  Reviews per Month  Host Listings Count  Future Availability  \
0              0.03                    1                  267   
1              0.73                    1                    0   
2               0.3                    2                  322   
3              3.38                    1                   79   
4              0.31                    3                  365   

   Reviews in Last 12 Months  
0                          1  
1                          0  
2                          0  
3                         50  
4                          1

## 4. Export the Cleaned Dataset

In [108]:
# Export the cleaned dataset to a csv file
df_list.to_csv(os.path.join(path, '02 Data','Prepared Data', 'Airbnb_cleaned.csv'))